petastorm/spark/spark_dataset_converter.py

```diff
@@ -23,72 +26,93 @@
 }
 
 
-class tf_dataset_context_manager:
-
-    def __init__(self, converter):
-        self.converter = converter
-
-    def __enter__(self) -> tf.data.Dataset:
-        return self.converter.dataset
-
-    def __exit__(self, exc_type, exc_value, exc_traceback):
-        self.converter.close()
-
-
 class SparkDatasetConverter:
     """
     The SparkDatasetConverter class manages the intermediate files when converting a SparkDataFrame to a
     Tensorflow Dataset or PyTorch DataLoader.
     """
-    def __init__(self, cache_file_path):
+    def __init__(self, cache_file_path: str, unischema: Unischema = None):
         """
         :param cache_file_path: The path to store the cache files.
+        :param unischema: Unischema that contains metadata of the dataset. TODO use this param.
         """
         self.cache_file_path = cache_file_path
+        self.unischema = unischema
         self.dataset = None
 
-    def make_tf_dataset(self) -> tf_dataset_context_manager:
+    def make_tf_dataset(self):
         reader = make_batch_reader("file://" + self.cache_file_path)
         self.dataset = make_petastorm_dataset(reader)
         return tf_dataset_context_manager(self)
 
     def close(self):
         """
-        Todo: delete cache files
+        Close the underlying reader. It would be called by the context manager at exit.
         :return:
         """
         pass
 
+    def delete(self):
+        """
+        Delete cache files at self.cache_file_path.
+        :return:
+        """
+        shutil.rmtree(self.cache_file_path, ignore_errors=True)
+
+
+class tf_dataset_context_manager:
+
+    def __init__(self, converter: SparkDatasetConverter):
+        self.converter = converter
+
+    def __enter__(self) -> tf.data.Dataset:
+        return self.converter.dataset
+
+    def __exit__(self, exc_type, exc_value, exc_traceback):
+        self.converter.close()
+
+
+def _get_uuid():
+    """
+    Generate a UUID from a host ID, sequence number, and the current time.
+    :return: a string of UUID.
+    """
+    return str(uuid.uuid1())
```

Comment:
use `uuid4()` ?

Reply:
uuid4() does not consider host address and the current time. Do you think uuid1 is better? Maybe there are no big differences but uuid4() is simpler and faster?

Reply:
I don't know, seemingly uuid4 is used more widely..

Reply:
https://www.sohamkamani.com/blog/2016/10/05/uuid1-vs-uuid4/ From this blog, I think we can just use `uuid4()` since we don't specifically want to put in host address and time information in the id, and the chances of collision for `uuid4()` are practically none.